test/apps/mmap/mmap_beyond_the_file.c

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MPL-2.0
+
+#include <sys/mman.h>
+#include <sys/fcntl.h>
+#include <unistd.h>
+
+#include "../test.h"
+
+#define PAGE_SIZE 4096
+
+FN_TEST(mmap_beyond_the_file)
+{
+	const char *filename = "mmap_test_file";
+	int fd = TEST_SUCC(open(filename, O_CREAT | O_RDWR, 0600));
+	TEST_SUCC(ftruncate(fd, 2 * PAGE_SIZE));
+
+	char *addr = CHECK_WITH(mmap(NULL, 4 * PAGE_SIZE,
+				     PROT_READ | PROT_WRITE, MAP_SHARED, fd, 0),
+				_ret != MAP_FAILED);
+	TEST_RES(addr[1 * PAGE_SIZE], _ret == 0);
+	// The following operation (if uncommented) would cause a segmentation fault.
+	// TEST_RES(addr[3 * PAGE_SIZE], _ret == 0);
+
+	TEST_SUCC(munmap(addr, 4 * PAGE_SIZE));
```

Comment:
Please also close and unlink the file, although it does not seem to be very important.
https://github.com/asterinas/asterinas/blob/5e3e23bf7c5130a9251b265c5377d2e21769097c/test/apps/mmap/mmap_and_mremap.c#L169-L170

Otherwise LGTM too. Let @junyang-zh make the final decision to merge this PR after addressing the above issues, as he is the correct code owner.